src/client/content_cao.cpp

```diff
@@ -1015,29 +1015,32 @@ void GenericCAO::step(float dtime, ClientEnvironment *env)
 			updateTextures(m_previous_texture_modifier);
 		}
 	}
-	if (!getParent() && std::fabs(m_prop.automatic_rotate) > 0.001) {
-		m_rotation.Y += dtime * m_prop.automatic_rotate * 180 / M_PI;
-		rot_translator.val_current = m_rotation;
-		updateNodePos();
-	}
 
-	if (!getParent() && m_prop.automatic_face_movement_dir &&
-			(fabs(m_velocity.Z) > 0.001 || fabs(m_velocity.X) > 0.001)) {
-		float target_yaw = atan2(m_velocity.Z, m_velocity.X) * 180 / M_PI
-				+ m_prop.automatic_face_movement_dir_offset;
-		float max_rotation_per_sec =
-				m_prop.automatic_face_movement_max_rotation_per_sec;
+	if (!getParent()) {
+		// This is the child node's rotation. It is only used for automatic_rotate.
+		v3f local_rot = node->getRotation();
```

Comment:
If it's only for `automatic_rotate`, please re-add the check to skip this part when it's not needed.
-> `if (std::fabs(m_prop.automatic_rotate) > 0.001) { ....`

Reply:
@SmallJoker Is this the only thing that needs to be attended to before the PR is ready for review again? I'll check the IRC chat logs to see if we decided to do something different regarding this PR that day.

(Apologies, I've gotten a little forgetful, the past couple of weeks)

Reply:
Yes, I think so. The current code skips `automatic_face_movement_dir` entirely when there's no auto rotation - a regression.

Reply:
Reverted this change. Sorry for the delay.